models/autoencoders/dvaepp.py

```diff
@@ -69,14 +69,15 @@ def forward(self, x):
         out=self._output_container.clear()
       	
 	    #TODO data prep - study if this does good things
-        input_data_centered=x.view(-1, self._flat_input_size)-self._dataset_mean
+        input_data_centered=x.view(-1, self._flat_input_size)#-self._dataset_mean
```

Comment:
This data prep was following the quadrant implementation. I had found a paper on centering RBM input data, which seemed to help with training. Did you try something else here and in the following with the train bias?

Reply:
The issue here is the different devices the two tensors are on. Easy hack is to use .to(x.device). Will add/